### PR TITLE
tokocrypto.createOrder string math

### PR DESCRIPTION
**Pull request by samgermain**
## JS

```
% tokocrypto createOrder ADA/USDT market buy 30 0.394    
2023-02-09T01:18:18.604Z
Node.js: v18.4.0
CCXT v2.7.32
tokocrypto.createOrder (ADA/USDT, market, buy, 30, 0.394)
2023-02-09T01:18:25.837Z iteration 0 passed in 2981 ms

{
  info: {
    orderId: '165404180',
    clientId: '675873107526189056',
    symbol: 'ADA_USDT',
    symbolType: '1',
    side: '0',
    type: '2',
    price: '0',
    origQty: '0',
    origQuoteQty: '11.82',
    executedQty: '29.90000000',
    executedPrice: '0.3941',
    executedQuoteQty: '11.78359000',
    timeInForce: '1',
    stopPrice: '0',
    icebergQty: '0',
    status: '2',
    isWorking: '1',
    createTime: '1675905505952',
    borderId: '3844196142',
    borderListId: '0'
  },
  id: '165404180',
  clientOrderId: undefined,
  timestamp: 1675905505952,
  datetime: '2023-02-09T01:18:25.952Z',
  lastTradeTimestamp: undefined,
  symbol: 'ADA/USDT',
  type: '2',
  timeInForce: '1',
  postOnly: false,
  reduceOnly: undefined,
  side: 'buy',
  price: undefined,
  stopPrice: undefined,
  triggerPrice: undefined,
  amount: 29.9,
  cost: undefined,
  average: undefined,
  filled: 29.9,
  remaining: 0,
  status: 'closed',
  fee: undefined,
  trades: [],
  fees: []
}
2023-02-09T01:18:25.837Z iteration 1 passed in 2981 ms
```

## Python

```
 % python3 examples/py/cli.py tokocrypto createOrder ADA/USDT market buy 30 0.394
Python v3.10.4
CCXT v2.7.32
tokocrypto.createOrder(ADA/USDT,market,buy,30,0.394)
{'amount': 29.9,
 'average': None,
 'clientOrderId': None,
 'cost': None,
 'datetime': '2023-02-09T01:20:08.873Z',
 'fee': None,
 'fees': [],
 'filled': 29.9,
 'id': '165404279',
 'info': {'borderId': '3844197163',
          'borderListId': '0',
          'clientId': '675873539188666368',
          'createTime': '1675905608873',
          'executedPrice': '0.3946',
          'executedQty': '29.90000000',
          'executedQuoteQty': '11.79854000',
          'icebergQty': '0',
          'isWorking': '1',
          'orderId': '165404279',
          'origQty': '0',
          'origQuoteQty': '11.82',
          'price': '0',
          'side': '0',
          'status': '2',
          'stopPrice': '0',
          'symbol': 'ADA_USDT',
          'symbolType': '1',
          'timeInForce': '1',
          'type': '2'},
 'lastTradeTimestamp': None,
 'postOnly': False,
 'price': None,
 'reduceOnly': None,
 'remaining': 0.0,
 'side': 'buy',
 'status': 'closed',
 'stopPrice': None,
 'symbol': 'ADA/USDT',
 'timeInForce': '1',
 'timestamp': 1675905608873,
 'trades': [],
 'triggerPrice': None,
 'type': '2'}
 ```

### CODE REVIEW

File: js/tokocrypto.js [hunk 1]
```diff
@@ -1607,7 +1608,10 @@ module.exports = class tokocrypto extends Exchange {
                     request['quoteOrderQty'] = this.decimalToPrecision (quoteOrderQty, TRUNCATE, precision, this.precisionMode);
                     params = this.omit (params, [ 'quoteOrderQty', 'cost' ]);
                 } else if (price !== undefined) {
-                    request['quoteOrderQty'] = this.decimalToPrecision (amount * price, TRUNCATE, precision, this.precisionMode);
+                    const amountString = amount.toString ();
```
Comment (kroitor) [inline js/tokocrypto.js:1611]: We can't use `toString()` here for reasons explained in the CONTRIBUTING doc: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

Screenshot 2023-02-07 at 19 44 40

Let's replace it with `this.numberToString (...)`.

**Comment by carlosmiei** [comment]
[🤖 TestsManager 🤖  Id: 63e44987c391f053efe0a9d2]
 ```

 Keys not found for tokocrypto, skipping tests!
```

**Comment by samgermain** [comment]
> 

doesn't that just mean that you don't have apiKeys?

**Comment by samgermain** [comment]
reopened in https://github.com/ccxt/ccxt/pull/18413

**Pull request closed without merging**